nmostesting/suites/IS1301Test.py

```diff
@@ -0,0 +1,316 @@
+import copy
+import time
+
+ANNOTATION_API_KEY = "annotation"
+NODE_API_KEY = "node"
+
+RESOURCES = ["self", "devices", "senders", "receivers"]
+OBJECTS = ["label", "description", "tags"]
+
+# const for label&description-related tests
+STRING_OVER_MAX_VALUE = ''.join(['X' for i in range(100)])
+STRING_MAX_VALUE = STRING_OVER_MAX_VALUE[:64]  # this is the max length tolerated
+
+# const for tags-related tests
+TAGS_OVER_MAX_VALUE = {'location': ['underground'], 'studio': ['42'], 'tech': ['John', 'Mike']}
+TAGS_MAX_VALUE = TAGS_OVER_MAX_VALUE.copy()
+TAGS_OVER_MAX_VALUE.pop('tech')  # must have a max of 5
+TAGS_TO_BE_SKIPPED = 'urn:x-nmos:tag:grouphint/v1.0'
+
+
+def get_ts_from_version(version):
+    """ Convert the 'version' object (string) into float """
+    return float(re.sub(':', '.', version))
+
+
+def strip_tags(tags):
+    if TAGS_TO_BE_SKIPPED in list(tags.keys()):
+        tags.pop(TAGS_TO_BE_SKIPPED)
+    return tags
+
+
+class IS1301Test(GenericTest):
+    """
+    Runs IS-13-Test
+    """
+
+    def __init__(self, apis, **kwargs):
+        GenericTest.__init__(self, apis, **kwargs)
+        self.annotation_url = self.apis[ANNOTATION_API_KEY]["url"]
+        self.node_url = f"{self.apis[ANNOTATION_API_KEY]['base_url']}/x-nmos/node/{self.apis[NODE_API_KEY]['version']}/"
+
+    def set_up_tests(self):
+        """
+        FAKE_ORIG = {
+            'description': 'fake_orig_desc',
+            'label': 'fake_orig_label',
+            'tags': {
+                'location': ['fake_location']
+            }
+        }
+        for resource in RESOURCES:
+            url = "{}{}{}".format(self.annotation_url, 'node/', resource)
+            TestHelper.do_request("PATCH", url, json=FAKE_ORIG)
+        """
+        pass
+
+    def get_resource(self, url):
+        """ Get a resource """
+
+        valid, r = self.do_request("GET", url)
+        if valid and r.status_code == 200:
+            try:
+                return True, r.json()
+            except Exception as e:
+                return False, e.msg
+        else:
+            return False, "GET  Resquest FAIL"
+
+    def compare_resource(self, object, new, resp):
+        """ Compare string values (or "tags" dict) """
+
+        if new[object] is None:  # this is a reset, new is null, skip
+            return True, ""
+
+        if object == "tags":  # tags needs to be stripped
+            resp[object] = strip_tags(resp[object])
+            if not TestHelper.compare_json(resp[object], new[object]):
+                return False, f"{object} value FAIL"
+
+        elif resp[object] != new[object]:
+            return False, f"{object} value FAIL"
+
+        return True, ""
+
+    def set_resource(self, url, node_url, new, prev):
+        """ Patch a resource with one ore several object values """
+
+        object = list(new.keys())[0]
+        valid, resp = TestHelper.do_request("PATCH", url, json=new)
+        if not valid:
+            return False, "PATCH Resquest FAIL"

```

Comment:
Rather than pass back a flag and message, you could raise `NMOSTestException`s in this function. (See `do_receiver_put` in `IS0401Test.py` for some examples of wrapping the test failure in this exception).  You could then remove the flag handling code in `do_test` making it more compact and readable.